### PR TITLE
Add metrics about Matrix API calls

### PR DESCRIPTION
**Pull request by jaller94**
This is an alternative to #67.

This adds the following Prometheus metrics:

* `bridge_matrix_api_calls` Counter: Number of API calls made
* `bridge_matrix_api_calls_failed` Counter: Number of API calls which failed

The names and types have been chosen to match those of https://github.com/matrix-org/matrix-appservice-bridge.

## Incomplete list of things which could be improved
- Add a meaningful `method` labels to other parts of the API, not just `events.py`.

## Differences to PR #67
- This covers all Matrix API calls, not just those in `events.js`, however, all others have an empty string as the `method` label.
- This counts retries as separate API calls. #67 considers retries to be part of the first call.
- Removed `bridge_matrix_request_seconds` s it was implemented differently from the NodeJS bridges.

### CODE REVIEW

**Comment by Half-Shot** [comment]
N.B. I *think* `matrix_request_seconds` is `bridge_matrix_request_seconds` because the bridge lib prefixes "bridge".

**Review by tulir — APPROVED** [review]

File: mautrix/api.py [hunk 1]
```diff
@@ -20,11 +20,18 @@
...
 from mautrix import __version__ as mautrix_version
 
 if TYPE_CHECKING:
     from mautrix.types import JSON
 
+API_CALLS = Counter("bridge_matrix_api_calls",
+                    "The number of Matrix client API calls made", ("method",))
+API_CALLS_FAILED = Counter("bridge_matrix_api_calls_failed",
+                           "The number of Matrix client API calls which failed", ("method",))
+MATRIX_REQUEST_SECONDS = Histogram("bridge_matrix_request_seconds",
```
Comment (Half-Shot) [inline mautrix/api.py:33]: In practise I think we got this wrong.

Comment (jaller94) [reply mautrix/api.py:33]: The only incorrect metric is `bridge_matrix_request_seconds`, right?

Comment (Half-Shot) [reply mautrix/api.py:33]: Yeah, the others look fine!

Comment (jaller94) [reply mautrix/api.py]: I removed `bridge_matrix_request_seconds` from this PR. I may come in another PR.

**Review by tulir — APPROVED** [review]

**Pull request merged**